### PR TITLE
Feature/39 | Added Auto Scaling Missing Security Groups plugin, spec file

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/autoscaling/asgMissingSecurityGroups.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+                    config.SecurityGroups.forEach(function(group){
+                        if (!securityGroups.includes(group)) {
+                            missingSecurityGroups.push(group);
+                        }
+                    });
+
+                    if(missingSecurityGroups.length) {
+                        helpers.addResult(results, 2,
+                            `Auto Scaling launch configuration "${config.LaunchConfigurationName}" refereces these missing EC2 security groups: ${missingSecurityGroups.join(', ')}`,
```
Comment (giorod3) [inline plugins/aws/autoscaling/asgMissingSecurityGroups.js:72]:
```suggestion
                            `Auto Scaling launch configuration "${config.LaunchConfigurationName}" references these missing EC2 security groups: ${missingSecurityGroups.join(', ')}`,
```

File: plugins/aws/autoscaling/asgMissingSecurityGroups.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    securityGroups.push(sg.GroupId);
+                });
+            }
+
+            describeLaunchConfigurations.data.forEach(function(config){
+                var resource = config.LaunchConfigurationARN;
+
+                if (!config.SecurityGroups || !config.SecurityGroups.length) {
+                    helpers.addResult(results, 2,
+                        `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not have any security group associated`,
```
Comment (giorod3) [inline plugins/aws/autoscaling/asgMissingSecurityGroups.js:58]:
```suggestion
                        `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not have any security groups associated`,
```

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/autoscaling/asgMissingSecurityGroups.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                        }
+                    });
+
+                    if(missingSecurityGroups.length) {
+                        helpers.addResult(results, 2,
+                            `Auto Scaling launch configuration "${config.LaunchConfigurationName}" references these missing EC2 security groups: ${missingSecurityGroups.join(', ')}`,
+                            region, resource);
+                    } else {
+                        helpers.addResult(results, 0,
+                            `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not reference any missing EC2 security group`,
```
Comment (giorod3) [inline plugins/aws/autoscaling/asgMissingSecurityGroups.js:76]:
```suggestion
                            `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not reference any missing EC2 security groups`,
```

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/autoscaling/asgMissingSecurityGroups.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                });
+            }
+
+            describeLaunchConfigurations.data.forEach(function(config){
+                var resource = config.LaunchConfigurationARN;
+
+                if (!config.SecurityGroups || !config.SecurityGroups.length) {
+                    helpers.addResult(results, 2,
+                        `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not have any security groups associated`,
+                        region, resource);
```
Comment (matthewdfuller) [inline plugins/aws/autoscaling/asgMissingSecurityGroups.js:59]:
```suggestion
                    helpers.addResult(results, 0,
                        `Auto Scaling launch configuration "${config.LaunchConfigurationName}" does not have any security groups associated`,
                        region, resource);
```

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
Might need to update the test for the new pass condition

**Review by giorod3 — APPROVED** [review]

**Pull request merged**